core/autoscaling.go

```diff
@@ -88,18 +117,78 @@ func (a *autoScalingGroup) loadConfOnDemand() bool {
 					return done
 				}
 			}
+		}
+		debug.Println("Couldn't find tag", tagKey)
+	}
+	return false
+}
+
+func (a *autoScalingGroup) loadBiddingPolicy(tagValue *string) (string, bool) {
+	biddingPolicy := *tagValue
+	if biddingPolicy != "aggressive" {
+		return DefaultBiddingPolicy, false
+	}
+
+	logger.Printf("Loaded BiddingPolicy value with %s from tag %s\n", biddingPolicy, BiddingPolicyTag)
+	return biddingPolicy, true
+}
+
+func (a *autoScalingGroup) loadConfSpot() bool {
+	tagList := []string{BiddingPolicyTag}
+	loadDyn := map[string]func(*string) (string, bool){
+		BiddingPolicyTag: a.loadBiddingPolicy,
+	}
+
+	for _, tagKey := range tagList {
+		if tagValue := a.getTagValue(tagKey); tagValue != nil {
+			if _, ok := loadDyn[tagKey]; ok {
+				if newValue, done := loadDyn[tagKey](tagValue); done {
+					a.region.conf.BiddingPolicy = newValue
+					logger.Println("BiddingPolicy =", a.region.conf.BiddingPolicy)
+					return done
+				}
+			}
 		} else {
 			debug.Println("Couldn't find tag", tagKey)
 		}
 	}
 	return false
 }
 
+func (a *autoScalingGroup) loadConfSpotPrice() bool {
+
+	if tagValue := a.getTagValue(SpotPriceBufferPercentageTag); tagValue != nil {
+		if newValue, done := a.loadSpotPriceBufferPercentage(tagValue); done {
+			if newValue > 0 {
+				a.region.conf.SpotPriceBufferPercentage = newValue
+				return done
+			}
+		}
+	} else {
+		debug.Println("Couldn't find tag", SpotPriceBufferPercentageTag)
+	}
+	return false
+}
```

Comment:
I believe this could be rewritten as:

```go
func (a *autoScalingGroup) loadConfSpotPrice() bool {

	tagValue := a.getTagValue(SpotPriceBufferPercentageTag)
	if tagValue == nil {
		return false
	}

	newValue, done := a.loadSpotPriceBufferPercentage(tagValue)
	if !done {
		debug.Println("Couldn't find tag", SpotPriceBufferPercentageTag)
		return false
	}

	a.region.conf.SpotPriceBufferPercentage = newValue
	return done
}
```

Which would also solve code climate issue I believe, as the comment about imbricated if/else is something that adds complexity to the code and that code climate dislike.